presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -236,9 +235,23 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
-        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
-        return getColumns(icebergTable.schema(), typeManager).stream()
-                .collect(toImmutableMap(IcebergColumnHandle::getName, identity()));
+        ImmutableList.Builder<IcebergColumnHandle> columns = new ImmutableList.Builder<>();
+        switch (table.getTableType()) {
+            case FILES:
+                IcebergTableHandle tblHandle = IcebergTableHandle.from(new SchemaTableName(table.getSchemaName(), table.getTableName()));
+                org.apache.iceberg.Table sourceTable = getIcebergTable(metastore, hdfsEnvironment, session, tblHandle);
+                FilesTable filesTable = new FilesTable(sourceTable.schema(), typeManager);
+                columns.addAll(filesTable.getColumnHandles());
+                break;
+            case DATA:
+                org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
+                columns.addAll(getColumns(icebergTable.schema(), typeManager));
+                break;
+            default:
+                throw new PrestoException(IcebergErrorCode.ICEBERG_UNKNOWN_TABLE_TYPE, "not supported" + table.getTableType());
```

Comment:
This error code means that it's not an Iceberg table (i.e. it's a Hive table instead). This error is actually a programming bug, so it would be fine to do
```java
throw new VerifyException("Unhandled table type: " + table.getTableType());